server/app/controllers/service-controller.js

```diff
@@ -25,27 +40,43 @@ async function getAllServices(req, res) {
  * @return {Promise.<void>} Nothing
  */
 async function createService(req, res) {
+
+
     const schema = Joi.array().items(ServiceSchema.requiredKeys(
-        'startingDate',
-        'endingDate',
+        'startAt',
+        'endAt',
         'nbMax',
         '_embedded.category' // If _embedded exist, there must be a category field
     )).min(1);
+    
 
     const { error } = schema.validate(req.body);
-    if (error) throw createUserError('BadRequest', error.details.message);
-
-    let newService = new Service({
-        ...req.body,
-        _embedded: undefined // Remove the only external object
-    });
-
-    newService = await serviceService.createService(newService, req.body._embedded);
-
-    res.json(newService);
+    if (error) throw createUserError('BadRequest', error.details[0].message);
+
+    //créer un tableau de services , a la place de new service etc.... (foreach...)
+    const services = new Array();
+
+    /* req.body.forEach(service => {
+        service = await serviceService.createService(service, service[_embedded]);
+        services.push(service);
+    });*/
+
+    for(let service of req.body) {
+        let newService = new Service({
+            name: service.name,
+            startAt: service.startAt,
+            endAt: service.endAt,
+            nbMax: service.nbMax
+        });
+        service = await serviceService.createService(newService, service._embedded);
```

Comment:
Don't do `await` in a `for` loop, (it's bad and will be prenvented soon :wink:).
See https://eslint.org/docs/rules/no-await-in-loop for more information.

ALso, I don't think we should have loop in the controller... What do you think @kistora ?